lib/widgets/emoji_reaction.dart

```diff
@@ -0,0 +1,325 @@
+        case ReactionType.realmEmoji:
+        case ReactionType.zulipExtraEmoji:
+          emoji = _ImageEmoji(
+            emojiCode: emojiCode,
+            emojiName: emojiName,
+            selected: selfVoted,
+          );
+      }
+    }
+
+    return Tooltip(
+      excludeFromSemantics: true, // TODO: Semantics with eg "Reaction: <emoji name>; you and N others: <names>"
+      message: emojiName,
+      child: Material(
+        color: backgroundColor,
+        shape: shape,
+        child: InkWell(
+          customBorder: shape,
+          splashColor: splashColor,
+          highlightColor: highlightColor,
+          onTap: () {
+            (selfVoted ? removeReaction : addReaction).call(store.connection,
+              messageId: messageId,
+              reactionType: reactionType,
+              emojiCode: emojiCode,
+              emojiName: emojiName,
+            );
+          },
+          child: Padding(
+            // 1px of this padding accounts for the border, which Flutter
+            // just paints without changing size.
+            padding: const EdgeInsetsDirectional.fromSTEB(4, 2, 5, 2),
+            child: Row(mainAxisSize: MainAxisSize.min, crossAxisAlignment: CrossAxisAlignment.center, children: [
+              Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                child: emoji),
+              Flexible(
+                // Added vertical: 1 to give some space when the label is
+                // taller than the emoji (e.g. because it needs multiple lines)
+                child: Padding(padding: const EdgeInsets.symmetric(horizontal: 3, vertical: 1),
+                  child: Text(
+                    textWidthBasis: TextWidthBasis.longestLine,
+                    textScaler: _labelTextScalerClamped(context),
+                    style: TextStyle(
+                      fontFamily: 'Source Sans 3',
+                      fontSize: (14 * 0.90),
+                      height: 13 / (14 * 0.90),
+                      color: labelColor,
+                    ).merge(selfVoted
+                      ? weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900)
+                      : weightVariableTextStyle(context)),
+                    label))),
+            ])))));
+  }
+}
+
+/// The size of the emoji square, or for :text_emojis:, the allotted height.
+///
+/// Should be scaled by [_emojiTextScalerClamped] for all emoji types.
+const _emojiSize = 17.0;
+
+/// A font size that, with Noto Color Emoji and our line-height config,
+/// causes a Unicode emoji to occupy a square of [_emojiSize] in the layout.
+///
+/// Determined experimentally:
+///   <https://github.com/zulip/zulip-flutter/pull/410#discussion_r1402808701>
+// TODO(#404) Actually bundle Noto Color Emoji with the app. Some newer Android
+//   phones seem to use Noto Color Emoji automatically, but it might not be
+//   universal, and at least the office Android device (which is old) doesn't.
```

Comment:
Yeah, this definitely isn't universal among Android devices, even new ones. In particular Samsung has their own emoji font.

See for example Emojipedia's list of glyphs for a random emoji:
https://emojipedia.org/moose#designs

They do say of Noto Color Emoji:
> Default emoji set for Android devices created by Google and used on devices by Google, Xiaomi, Oppo, Huawei, Sony, and many others.